### PR TITLE
chore(ci): bump github action versions

### PR DESCRIPTION
**Pull request by corneliusroemer**
- **chore(ci): bump actions/checkout to v4**
- **chore(ci): bump actions/cache to v4**
- **chore(ci): bump upload-artifact to v4**

Fixes lints in Github actions such as this:

Brave Browser 2024-07-17 12 03 17

### CODE REVIEW

File: .github/workflows/static_build.yml [hunk 1]
```diff
@@ -138,15 +134,7 @@ jobs:
...
-            cli11>=2.2,<3
-            cpp-expected
-            nlohmann_json
-            simdjson-static>=3.3.0
-            spdlog
-            fmt
-            yaml-cpp-static>=0.8.0
-            libsolv-static>=0.7.24
-            reproc-cpp-static>=14.2.4.post0
+            cli11>=2.2,<3 cpp-expected nlohmann_json simdjson-static>=3.3.0 spdlog fmt yaml-cpp-static>=0.8.0 libsolv-static>=0.7.24 reproc-cpp-static>=14.2.4.post0
```
Comment (Hind-M) [inline .github/workflows/static_build.yml:137]: Why not leave it as it is? It's more readable and it complies with the the examples in [setup-micromamba](https://github.com/mamba-org/setup-micromamba).

Comment (corneliusroemer) [reply .github/workflows/static_build.yml:137]: Oh this might have been my yaml formatter - sorry for that!

Comment (corneliusroemer) [reply .github/workflows/static_build.yml]: Fixed it

File: .github/workflows/static_build.yml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 name: Micromamba
-
```
Comment (Hind-M) [inline .github/workflows/static_build.yml:2]: I think we should keep the space here for consistency with the rest of the yaml files 🙏 (must be the linter I guess).

File: .github/workflows/static_build.yml [hunk 3]
```diff
@@ -14,11 +13,9 @@ on:
       - '**.md'
   merge_group:
     types: [checks_requested]
-
```
Comment (Hind-M) [inline .github/workflows/static_build.yml:17]: Same.

File: .github/workflows/static_build.yml [hunk 4]
```diff
@@ -14,11 +13,9 @@ on:
...
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
-
```
Comment (Hind-M) [inline .github/workflows/static_build.yml:21]: Same.

File: .github/workflows/static_build.yml [hunk 5]
```diff
@@ -98,25 +98,24 @@ jobs:
...
           path: ${{ github.workspace }}/micromamba-conda-build-failed-${{ matrix.platform }}-${{ matrix.arch }}.tar.gz
           retention-days: 7
       - name: Upload micromamba
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: micromamba-${{ matrix.platform }}-${{ matrix.arch }}
           path: ${{ github.workspace }}/artifacts/micromamba
-
   micromamba-static-win:
```
Comment (corneliusroemer) [inline .github/workflows/static_build.yml:112]: Here this seems like random whitespace - it's not a top level key

**Review by Hind-M — APPROVED** [review]
Thanks!

**Pull request merged**